abstract/group.go

```diff
@@ -51,6 +51,9 @@ type Secret interface {
 
 	// Set to a fresh random or pseudo-random secret
 	Pick(rand cipher.Stream) Secret
+
+	// SetBytes will take bytes and create a secret out of it
+	SetBytes([]byte) Secret
```

Comment:
Wait, what?  From what I'm reading, it sounds like you're saying the exact opposite of what I thought the semantics should be. :)

- MarshalBinary() should ALWAYS produce exactly MarshalSize() bytes, independent of the value of the integer being marshaled.  That's why it has the code above to pad the return from i.V.Bytes() if it is too short.

- Bytes() should probably behave the same way as big.Int.Bytes(), namely returning a byte-slice just large enough to hold the actual integer being encoded.  

In other words, SetBytes() takes a variable-length integer of any size as input, and Bytes() produces a variable-length return.  UnmarshalBinary() always consumes *exactly* MarshalSize() bytes, and MarshalBinary() always produces *exactly* MarshalSize() bytes.

So the implementation of Bytes() in this module should just call i.v.Bytes(), then reverse the slice if the caller wants little-endian.

Make sense?